private/bufpkg/bufmodule/bufmoduleref/validate.go

```diff
@@ -59,18 +57,9 @@ func ValidateProtoModulePin(protoModulePin *modulev1alpha1.ModulePin) error {
 	if err := ValidateRepository(protoModulePin.Repository); err != nil {
 		return err
 	}
-	if err := ValidateBranch(protoModulePin.Branch); err != nil {
```

Comment:
I didn't remove the function because there are currently some server-side use-cases I am currently unsure of and wanted to discuss with @WillAbides in relation to his work on tracks.

Reply:
Sounds good.

For the sake of completeness here, is it possible to remove _all_ client-side validation? It looks like we're still including `ValidateProtoModulePin` as a whole, so we would still be validating against remotes, owners, repositories, etc. 

If we ever decided to drop any of these fields from the ModulePin, then old CLIs would break. I know it's unlikely since these fields feel the most minimal, but I'd challenge us to see how much validation we can remove (without comprising basic functionality) to prevent these types of failures later.